compendium/ServiceComponent/include/cppmicroservices/servicecomponent/detail/ComponentInstanceImpl.hpp

```diff
@@ -481,11 +481,11 @@ namespace cppmicroservices
                               class I = Injection,
                               class InjectionTrue = typename std::enable_if<I::value == true>::type,
                               class HasConstructorWithReferences = typename std::enable_if<
-                                  std::is_constructible<C, std::shared_ptr<CtorInjectedRefs> const&...>::value>::type>
```

Comment:
So since we have 2 types now, shared_ptr for unary and vector<shared_ptr> for multiple, we needed a way for these to be passed to ComponentInstanceImpl such that they are deduced at compile time as unary/multiple and we can further invoke the GetDependencies code then. I have removed shared_ptr in above code as now the template parameters of this class will go as shared_ptr<ref_type> for unary and  vector<shared_ptr<ref_type>> instead of just ref_types.